dropwizard-client/src/test/java/io/dropwizard/client/HttpClientBuilderTest.java

```diff
@@ -679,6 +680,19 @@ public void usesHttpProcessor() throws Exception {
     }
 
     @Test
+    public void usesServiceUnavailableRetryStrategy() throws Exception {
+        ServiceUnavailableRetryStrategy serviceUnavailableRetryStrategy = mock(ServiceUnavailableRetryStrategy.class);
+        final ConfiguredCloseableHttpClient client =
+            builder.using(serviceUnavailableRetryStrategy)
+                .createClient(apacheBuilder, connectionManager, "test");
+        assertThat(client).isNotNull();
+        assertThat(FieldUtils.getField(httpClientBuilderClass,
+            "serviceUnavailStrategy", true)
```

Comment:
This part was quite tricky, as internally it's named quite inconsistently :cry:
But I guess this is a punishment for forcing access to the field via reflection :wink:

Reply:
I am also not a fan of unit tests with Reflection, especially considering most of them will fail on the new builds of Java 9, but in this case it should be good to go. Let's hope Apache Http Client will provide public accessors in future versions.